sml/asyncio.py

```diff
@@ -126,10 +133,21 @@ async def connect(self, loop=None):
         await self._reconnect(delay=0)
 
     async def _disconnect(self):
+        if self._watchdog and not self._watchdog.done():
+            self._watchdog.cancel()
         if self._transport:
             self._transport.abort()
             self._transport = None
 
+    async def _timeout(self):
+        while True:
+            await asyncio.sleep(self._timeout_delay)
```

Comment:
I was thinking about something like this:
```python3
sleep_time = self._last_update + self._timeout_delay - time.time()
if sleep_time > 0:
    await asyncio.sleep(sleep_time)
```
or 
```python3
last_update = self._last_update
sleep_time = last_update + self._timeout_delay - time.time()
if sleep_time > 0:
    await asyncio.sleep(sleep_time)
    if last_update != self._last_update:
        continue
logger.warning("Timeout ...")
```

Reply:
I really like the second version. But with this i need to re-set last_update everytime the connection is made. The reason why it was hardcoded initially is, that after a reconnect it doesn't get reconnected immediately. 
With the current commit it should work as expected and also uses your second suggestion. Thank you!